privacyidea/lib/tokens/vascotoken.py

```diff
@@ -157,9 +157,13 @@ def check_otp(self, otpval, counter=None, window=None, options=None):
                 # wrong OTP value, no log message
                 pass
             elif result == 201:
-                log.warning("A previous OTP value was used again!")
+                log.warning("VASCO token failed to authenticate, code replay attempt, previous OTP value was used again!")
             elif result == 202:
                 log.warning("Token-internal fail counter reached its maximum!")
+            elif result == -202:
```

Comment:
Is it really ``-202``? It was ``202`` before!
The ``result`` is returned from our ``vasco_otp_check``. Is the minus maybe dropped?

Reply:
Yes according to the vasco documentation, and to the current privacyidea.log output we are having entries like this:

[WARNING][privacyidea.lib.tokens.vascotoken:164] VASCO token failed to authenticate, result: -205
[WARNING][privacyidea.lib.tokens.vascotoken:164] VASCO token failed to authenticate, result: -202

I'm actually not sure what the positive 202 error is. Can't find this in the documentation.
And also can't find any log output with: "Token-internal fail counter reached its maximum" in our logs.

Reply:
Ah, ok, you are not touching it anyway.
It helps to be able to read the diff view! :-)